resource-aggregate/cqrs/eventstore/mongodb/eventstore.go

```diff
@@ -271,234 +299,317 @@ func (s *EventStore) ensureIndex(ctx context.Context, col *mongo.Collection, ind
 	return nil
 }
 
-func getEventCollectionName(groupID string) string {
-	return eventCName + "_" + groupID
+func getEventCollectionName() string {
+	return "devices_" + eventCName
+}
+
+func getDocID(event eventstore.Event) string {
+	return fmt.Sprintf("%v.%v", event.AggregateID(), event.Version())
+}
+
+func makeDBDoc(events []eventstore.Event, marshaler MarshalerFunc) (bson.M, error) {
+	e, err := makeDBEvents(events, marshaler)
+	if err != nil {
+		return nil, fmt.Errorf("cannot insert first events('%v'): %w", events, err)
+	}
+	return bson.M{
+		idKey:            getDocID(events[0]),
+		groupIDKey:       events[0].GroupID(),
+		aggregateIDKey:   events[0].AggregateID(),
+		latestVersionKey: events[len(events)-1].Version(),
+		firstVersionKey:  events[0].Version(),
+		isCurrentKey:     true,
+		eventsKey:        e,
+	}, nil
 }
 
 // Save save events to path.
-func (s *EventStore) Save(ctx context.Context, collectionID, aggregateID string, events []eventstore.Event) (concurrencyException bool, err error) {
+// All events must have same AggregateID and GroupID.
+// All events must have filled AggregateID, GroupID and Eventype.
+// Versions among events must continues ascending.
```

Comment:
update this also here.
And please document behaviour and requirements for event store also in the documentation.

Reply:
updated. I m not sure what you meaning with document behaviour.